test/9.custom-request-clients.test.js

```diff
@@ -0,0 +1,119 @@
+/* global describe, it */
+'use strict'
+
+const request = require('supertest')
+const expect = require('chai').expect
+
+let service
+
+describe('fast-proxy custom request clients', () => {
+  it('init & start remote service', async () => {
+    // init remote service
+    service = require('restana')()
+
+    service.get('/service/redirect', (req, res) => {
+      res.setHeader('location', '/service/redirected')
+      res.send(301)
+    })
+    service.get('/service/redirected', (req, res) => {
+      res.setHeader('url', req.url)
+      res.send()
+    })
+
+    await service.start(3000)
+  })
+
+  describe('with default request clients', () => {
+    let gateway, close, proxy, gHttpServer
+
+    it('init proxy', async () => {
+      const fastProxy = require('../index')({
+        base: 'http://127.0.0.1:3000'
+      })
```

Comment:
@jkyberneees By the way, within some of your tests you're [using `pump(stream, res)` within an `opts.onResponse`](https://github.com/fastify/fast-proxy/blob/9cfc1a7c98a010ad7992fed304a87655946a9242/test/4.opts.test.js#L39-L41). I removed this because it was causing the proxy to respond with a `200 OK` status code no matter what -- even when I tried to hit URLs that did not even exist. Very confusing!

Reply:
This is totally fine for the `options` test cases. This is not the case in other tests.